### PR TITLE
upgrade major runtime libraries

### PR DESCRIPTION
**Pull request by owen26**
Major changes including:

- react from 16 to 18, upgrade initialisation component and update explicit `children` prop
- blockly from 6 to 10, make a lot of breaking API changes compatible
- redux toolkit from 1.7 to 2.1
- js-interpreter from 2 to 5
- react-router-dom from 5 to 6, update routing setup with v6 breaking changes
- fix all lint errors
- remove google login dependencies and remove user/auth components as they are deprecated

### CODE REVIEW

**Comment by owen26** [comment]
manually tested, and would be great if someone else can also double check and do a quick manual testing before vs after the upgrade

File: src/BlocklyInterface/blocklySlice.ts [hunk 1]
```diff
@@ -138,7 +138,9 @@ export const saveBlocklyState = (state: RootState) => {
   try {
     const serializedState = JSON.stringify(state.blockly);
     localStorage.setItem(localStorageKey, serializedState);
-  } catch {}
+  } catch {
+    /* empty */
+  }
```
Comment (immutable-pro) [inline src/BlocklyInterface/blocklySlice.ts:143]: Note: I know this is pre-existent code :). 

`stringify` will only throw if it finds a circular dependency. And `setItem` will only throw if the storage quota is exceeded.  In any of the cases, something fishy might be happening with the app if this throws. So let's not consume it. 

```suggestion
  } catch (err) {
    console.error(err)
  }
```

Or, **even better**, let's not catch it and make the app nicely crash.

**Review by immutable-pro — APPROVED** [review]
Thank you, @owen26, for the update and maintenance. 
I have reviewed the code and tested it locally. 

Unless the changes were very inter-dependent, I think smaller PRs will facilitate a bit the reasoning about the changes, when dealing with so many libraries for different use cases. But it looks clean and everything seems to be working, as far as I understand the app.

File: src/BlocklyInterface/blocklySlice.ts (hunk 1 above)
Comment (owen26) [reply src/BlocklyInterface/blocklySlice.ts]: yup that make sense, I was just did some quick lint fixing but didn't put too much thinking around the implementation

**Comment by owen26** [comment]
Thanks @immutable-pro ! I promise this will be the last big PR I make 😂  So next time it'll be more reviewer friendly

**Pull request merged**